engine/src/test/java/io/camunda/zeebe/engine/processing/job/FailJobTest.java

```diff
@@ -315,4 +317,23 @@ public void shouldFailWithVariables() {
         .hasName("foo")
         .hasValue("\"bar\"");
   }
+
+  @Test
+  public void shouldTruncateErrorMessage() {
+    // given
+    final Record<JobRecordValue> job = ENGINE.createJob(jobType, PROCESS_ID);
+    final String exceedingErrorMessage = "*".repeat(501);
+
+    // when
+    final Record<JobRecordValue> failedRecord =
+        ENGINE.job().withKey(job.getKey()).withErrorMessage(exceedingErrorMessage).fail();
+
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(CREATED).getFirst();
+
+    // then
+    Assertions.assertThat(failedRecord).hasRecordType(RecordType.EVENT).hasIntent(FAILED);
+    assertThat(failedRecord.getValue().getErrorMessage().length()).isEqualTo(500);
+    assertThat(incident.getValue().getErrorMessage().length()).isEqualTo(500);
```

Comment:
🔧 We could assert the contents of the error message as well

Reply:
Updated. I removed the length assertions since we are asserting the exact string.